src/gsy_e/models/strategy/update_frequency.py

```diff
@@ -123,6 +128,7 @@ def delete_past_state_values(self, current_market_time_slot: DateTime) -> None:
             self.final_rate.pop(market_slot, None)
             self.energy_rate_change_per_update.pop(market_slot, None)
             self.update_counter.pop(market_slot, None)
+            self.market_slot_added_time_mapping.pop(market_slot, None)
```

Comment:
I am not sure if `delete_past_state_values` is even triggered for future markets. At least what I see is that dicts like `update_counter`, `initial_rate`, `market_slot_added_time_mapping` are growing in items endlessly. Do we miss `_delete_past_state` for future markets in `event_market_cycle` ? Or maybe there is other reason for such behavior?

Reply:
I have checked and you are right, the cleaning up of the past state seems to be missing.
I believe that the `_delete_past_state` should be overridden and adapted for the future markets.
Unless this issue is affecting the DoD of the bug ticket, I'll create a ticket to resolve this problem if you agree.